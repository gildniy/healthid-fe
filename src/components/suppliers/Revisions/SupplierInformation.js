import React, { Fragment } from 'react';
import { Typography, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';
import { FieldTitle, getFields } from './SupplierFields';
import capitalize from '../../utils/capitalize';

const SupplierInformation = (props) => {
  const {
    classes, supplier, currentSupplier
  } = props;
  const { supplierMeta } = supplier;
  const { suppliersmetaSet } = currentSupplier;

  const enteredFields = {
    tier: supplier.tier.name,
    paymentTerms: supplierMeta[0] && supplierMeta[0].paymentTerms
      .split('_').map(word => capitalize(word)).join(' '),
    creditDays: supplierMeta[0] && `${supplierMeta[0].creditDays} Days`
  };
  const currentFields = {
    tier: capitalize(currentSupplier.tier.name),
    paymentTerms: suppliersmetaSet[0] && suppliersmetaSet[0].paymentTerms
      .split('_').map(word => capitalize(word)).join(' '),
    creditDays: suppliersmetaSet[0] && `${suppliersmetaSet[0].creditDays} Days`
  };

  const FieldValues = getFields(currentFields, enteredFields, classes);
  const { fieldsChanged } = FieldValues;

  return (
    <Fragment>
      <div className={classes.dividerDiv}>
        <Typography
          paragraph
          align="left"
          gutterBottom
          className={classes.dividerHeaders}
        >
          Supplier Information
        </Typography>
      </div>

      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid item xs={4}>
          <FieldTitle changed={'tier' in fieldsChanged} title="Tier" classes={classes} />
          <FieldValues.Render name="tier" />
        </Grid>
        <Grid item xs={4}>
          <FieldTitle changed={'paymentTerms' in fieldsChanged} title="Payment Terms" classes={classes} />
          <FieldValues.Render name="paymentTerms" />
        </Grid>
        <Grid item xs={4}>
          <FieldTitle changed={'creditDays' in fieldsChanged} title="Credit Days" classes={classes} />
          <FieldValues.Render name="creditDays" />
        </Grid>
      </Grid>
    </Fragment>
  );
};

SupplierInformation.propTypes = {
  classes: PropTypes.shape({
    containerGrid: PropTypes.string,
    dividerHeaders: PropTypes.string,
    newTextFields: PropTypes.string,
    descriptionFields: PropTypes.string,
    dividerDiv: PropTypes.string,
  }).isRequired,
  supplier: PropTypes.instanceOf(Object).isRequired,
  currentSupplier: PropTypes.instanceOf(Object).isRequired
};

export default SupplierInformation;
