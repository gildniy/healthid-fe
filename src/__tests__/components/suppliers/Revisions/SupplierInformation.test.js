import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SupplierInformation from '../../../../components/suppliers/Revisions/SupplierInformation';

const props = {
  supplier: {
    tier: {
      name: 'Manufacturer'
    },
    supplierMeta: [{
      paymentTerms: 'ON_CREDIT',
      creditDays: 10
    }],
  },
  currentSupplier: {
    isApproved: true,
    tier: {
      name: 'Manufacturer'
    },
    suppliersmetaSet: [{
      paymentTerms: 'ON_CREDIT',
      creditDays: 10
    }],
  },
  renderTextField: jest.fn(),
  variables: {},
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: 'Africa/Nairobi'
          }
        }
      }
    }
  }
};

describe('SupplierInformation', () => {
  it('should render the SupplierInformation component properly', () => {
    const setupSupplierInformation = () => {
      const wrapper = shallow(<SupplierInformation {...props} />);
      return wrapper;
    };

    const wrapper = setupSupplierInformation();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
