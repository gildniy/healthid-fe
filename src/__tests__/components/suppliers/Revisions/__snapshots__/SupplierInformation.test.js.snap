// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierInformation should render the SupplierInformation component properly 1`] = `
<Fragment>
  <div>
    <WithStyles(ForwardRef(Typography))
      align="left"
      gutterBottom={true}
      paragraph={true}
    >
      Supplier Information
    </WithStyles(ForwardRef(Typography))>
  </div>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={24}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <FieldTitle
        changed={false}
        classes={Object {}}
        title="Tier"
      />
      <Render
        name="tier"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <FieldTitle
        changed={false}
        classes={Object {}}
        title="Payment Terms"
      />
      <Render
        name="paymentTerms"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <FieldTitle
        changed={false}
        classes={Object {}}
        title="Credit Days"
      />
      <Render
        name="creditDays"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
